overlord/ifacestate/ifacestate.go

```diff
@@ -203,8 +210,11 @@ func setInitialConnectAttributes(ts *state.Task, plugSnap string, plugName strin
 	if err != nil {
 		return err
 	}
+
+	emptyDynamicAttrs := make(map[string]interface{})
 	if plug, ok := snapInfo.Plugs[plugName]; ok {
-		ts.Set("plug-attrs", plug.Attrs)
+		ts.Set("plug-static", plug.Attrs)
```

Comment:
what happens on update with old tasks that had the previous thing?

Reply:
Ok, I thought about that and we're good. We're only reading static attributes in snapctl from interface hooks (but no one should be using them yet). And when it comes to dynamic attributes, with the change to getDynamicHookAttributes you suggested below we will ignore ErrNoState and use empty dynamic attributes. ets